kanisterio/kanister
# refactor code to remove extension group deps

## Change Overview



## Pull request type

Please check the type of change your PR introduces:

## Issues

- #XXX

## Test Plan
So, the issue was coming when I tried to run the stable/mysql example on k8s version `1.16`, and after creating the `backup actionset` the issue on the operators's pod was `Unable to fetch the replicaset`.  After the change that are in this PR I have tested the same example and followed below steps.
1. Run a mysql instance, install through helm.
2. Create a dummy database and table, and store some data into that table.
3. Create an actionset that will backup this table so that if anything bad happens to the database that we just created we would have back up to restore the data from.
4. I manually dropped the database and created the `restore actionset` to restore the backup that we already have.
5. Backup was successfully restored and we were able to see the data in the respective database.

By seeing the api reference we are sure that our actual code change is backword compatible at least till version `1.10` but if take `example` as well in consideration, ` examples/picture-gallery-pvc-snapshot/picture-gallery-deployment.yaml` is not going to work on k8s version less than `1.14`.
